### PR TITLE
Refactor: `quant_base_url` and `quant_hostname` usage for clarity and bug fix.

### PR DESCRIPTION
**Pull request by kepol**
See d.o issue below including last comment. Note the `@fixme` question. This code applies the fix from https://github.com/quantcdn/drupal/pull/167 more broadly.

https://www.drupal.org/project/quantcdn/issues/3366894

### CODE REVIEW

**Review by steveworley — APPROVED** [review]

File: modules/quant_api/quant_api.module
Comment (steveworley) [inline modules/quant_api/quant_api.module:186]: Yeah this should probably be `quant_get_base_url()` — but the local vs. remote identification should be a little more robust too. We should probably pull that out into a separate issue to refactor and optimise.

Comment (kepol) [reply modules/quant_api/quant_api.module:186]: okay... I can change it for now and make an issue to revisit local vs remote

**Comment by kepol** [comment]
I'll wait on this one before updating this as there will be merge conflicts and more places to update:

https://github.com/quantcdn/drupal/pull/154

**Review by steveworley — APPROVED** [review]

**Review by steveworley — APPROVED** [review]
Tested with test content and seeding works as expected:

- Tested pushing content with absolute URLs
- Tested pushing content with relative URLs
- Tested full site seed

All looks to be working I think this is ready to go.

**Pull request merged**
